coretex/coretex/sample/sequence_sample/local_sequence_sample.py

```diff
@@ -53,18 +69,83 @@ def sequencePath(self) -> Path:
             ------
             FileNotFoundError -> if no .fasta, .fastq, .fq, or .fq files are found inside the sample
         """
-        return getSequenceFile(Path(self.path), self.supportedExtensions())
+        return getSequenceFile(self.path, self.supportedExtensions())
+
+    @property
+    def forwardPath(self) -> Path:
+        """
+            Returns the path of the .fasta or .fastq forward sequence file
+            contained inside the sample. "_R1_" must be present in the filename
+            otherwise it will not be recongnized. If the sample contains gzip compressed
+            sequences, you will have to call Sample.unzip method first otherwise
+            calling Sample.sequencePath will raise an exception
+
+            Raises
+            ------
+            FileNotFoundError -> if no .fasta, .fastq, .fq, or .fq files are found inside the sample
+        """
+        return getForwardSequenceFile(self.path, self.supportedExtensions())
+
+    @property
+    def reversePath(self) -> Path:
+        """
+            Returns the path of the .fasta or .fastq sequence file
+            contained inside the sample. "_R2_" must be present in the filename
+            otherwise it will not be recongnized. If the sample contains gzip compressed
+            sequences, you will have to call Sample.unzip method first otherwise
+            calling Sample.sequencePath will raise an exception
+
+            Raises
+            ------
+            FileNotFoundError -> if no .fasta, .fastq, .fq, or .fq files are found inside the sample
+        """
+        return getReverseSequenceFile(self.path, self.supportedExtensions())
 
     def unzip(self, ignoreCache: bool = False) -> None:
         super().unzip(ignoreCache)
-
         try:
-            compressedSequencePath = getSequenceFile(Path(self.path), [".gz"])
-            decompressedSequencePath = compressedSequencePath.parent / compressedSequencePath.stem
+            for compressedSequencePath in self.path.iterdir():
```

Comment:
let's refactor this loop a bit, instead of using `iterdir`, loop through all `supportedExtensions()` and do `self.path.glob(f"*{supportedExtension}.gz")` and decompress every single file returned by the glob